### DISCUSSION TITLE
Ability to organize Checklists

### DISCUSSION THREAD

**Discussion by ErnestoPR** [Daniele-rolli/Beaver-Notes#108] [Ideas]
Hello. First of all thanks for the amazing app. I was wondering if it was possible to add the ability to sort a checklist. I use the app for notes but also as a To-Do app through the checklist option. If there are many items in the list and some are checked, then the list can feel a bit disorganized. It would be nice to have some sort of edit menu in which you can drag the checklist items up and down or the ability to sort the list by checked or unchecked to keep it organized.

**Comment by Daniele-rolli** [comment]
I see what you mean, however I think crossing out the items or reorganizing the bullet list so that items that are already checked go below, and the ones that haven't been checked will take their place would be a better option.

**Reply by ErnestoPR** [reply]
Thank you for your reply!. That works also. Anything that helps with the sorting. Another thing that can help is highlighting text and dragging it above or below where you want it and have it insert with the checkbox. Highlighting and moving text doesn't work and and empty space with a checkbox has to be manually created before cutting and pasting text lines.

**Reply by Daniele-rolli** [reply]
Okay, so I've been trying to work on a viable solution where checked items are much less visible than the unchecked ones, making it easier to browse through checklists. As far as dragging and dropping items in task lists is concerned, I believe it might be impossible since tiptap, the editor, handles lists as single items, meaning they cannot be dragged like code blocks or images. I hope this solution is acceptable; otherwise, we'll continue working together on something else.

![Screenshot 2024-02-06 at 7 18 56 PM](https://github.com/Daniele-rolli/Beaver-Notes/assets/67503004/5d01660b-e0c9-48f8-aa6d-8b44641221a9)

**Reply by ErnestoPR** [reply]
This actually helps a lot since I end up manually adding the strikethrough to them anyway. I looked around online and tried a few things with chatgpt but I don't have enough knowledge to help out. Sorry.

**Reply by Daniele-rolli** [reply]
I appreciate your willingness to help, and I'm happy we managed to sort out this issue.

**Comment by grand-lotus-iroh** [comment]
> ..drag the checklist items up and down or the ability to sort the list by checked or unchecked to keep it organized...

In [workflowy](https://blog.workflowy.com/tag/keyboard-shortcuts/), and other outliner apps, they use `Alt` + `Shift` + `Up`/ `Down` so that might be something cool to integrate for better sorting. I know ErnestoPR was talking about D&D but maybe a good starting point is hotkeys.

_something to consider_
